### ISSUE TITLE
Skins are not showing up correctly

### ISSUE THREAD

**Issue by suresttexas00** [suresttexas00/minecraft-wrapper#46]
I know this is hard to see in the picture, but each player sees the other with their own generic skin.  SurestTexas (steve based) sees two steves and SapperLeader2 (alex based) sees two alexes:

![image](https://cloud.githubusercontent.com/assets/10001762/14482914/d20330d4-010f-11e6-9218-ff3a2294bc1f.png)

console output:
```
surest@surest-Satellite-L55-B ~ $ cd Desktop/surest@surest-Satellite-L55-B ~/Desktop $ cd server\ test/surest@surest-Satellite-L55-B ~/Desktop/server test $ python "/home/surest/Desktop/server  
 ...

00:02:21] [Wrapper.py/DEBUG]: [Chat] Registered help group 'Chat' with 4 commands
[00:02:21] [Wrapper.py/INFO]: Starting server...
[00:02:24] [Server thread/INFO]: Starting minecraft server version 1.9.2
[00:02:24] [Server thread/WARN]: To start the server with more ram, launch it as "java -Xmx1024M -Xms1024M -jar minecraft_server.jar"
[00:02:24] [Server thread/INFO]: Loading properties
[00:02:24] [Server thread/WARN]: ressource-pack-hash is depricated and found along side resource-pack-sha1. resource-pack-hash will be ignored.
[00:02:24] [Server thread/INFO]: Default game type: SURVIVAL
[00:02:24] [Server thread/INFO]: Generating keypair
[00:02:24] [Server thread/INFO]: Starting Minecraft server on 127.0.0.1:25541
[00:02:24] [Server thread/INFO]: Using epoll channel type
[00:02:24] [Server thread/WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[00:02:24] [Server thread/WARN]: The server will make no attempt to authenticate usernames. Beware.
[00:02:24] [Server thread/WARN]: While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
[00:02:24] [Server thread/WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[00:02:24] [Server thread/INFO]: Preparing level "worldx"
[00:02:25] [Server thread/INFO]: Preparing start region for level 0
[00:02:26] [Server thread/INFO]: Preparing spawn area: 2%
[00:02:27] [Server thread/INFO]: Preparing spawn area: 68%
[00:02:27] [Server thread/INFO]: Done (2.727s)! For help, type "help" or "?"
[00:02:27] [Wrapper.py/INFO]: Server started
[00:02:51] [Server thread/INFO]: Turned off world auto-saving
[00:04:22] [Wrapper.py/INFO]: Starting new HTTPS connection (1): sessionserver.mojang.com
[00:04:22] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:04:22] [Wrapper.py/DEBUG]: "GET /session/minecraft/hasJoined?username=SurestTexas00&serverId=2ef61e4f8a10f4b63370924dded9ae2e49021c63 HTTP/1.1" 200 1125
[00:04:22] [Wrapper.py/INFO]: Starting new HTTPS connection (1): api.mojang.com
[00:04:22] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:04:22] [Wrapper.py/DEBUG]: "GET /user/profiles/4f95325567754dc6a612fb4230588eff/names HTTP/1.1" 200 26
[00:04:22] [Wrapper.py/INFO]: SurestTexas00 logged in (UUID: 4f953255-6775-4dc6-a612-fb4230588eff | IP: 68.50.227.172)
[00:04:22] [Server thread/INFO]: SurestTexas00[/127.0.0.1:60593] logged in with entity id 165 at (3344.223968901004, 67.0, 2791.2503706879593)
[00:04:22] [Server thread/INFO]: SurestTexas00 joined the game
Client.py - chatmsg: /help
Command.py - args: 
[]
Command.py - payload['args']: []
Command.py - ' '.join(payload['args']): 
[00:05:12] [Wrapper.py/INFO]: SurestTexas00 executed: /help 
Client.py - chatmsg: /help Toolbox
Command.py - args: Toolbox
['Toolbox']
Command.py - payload['args']: ['Toolbox']
Command.py - ' '.join(payload['args']): Toolbox
[00:05:17] [Wrapper.py/INFO]: SurestTexas00 executed: /help Toolbox
Client.py - chatmsg: /help VanillaClaims
Command.py - args: VanillaClaims
['VanillaClaims']
Command.py - payload['args']: ['VanillaClaims']
Command.py - ' '.join(payload['args']): VanillaClaims
[00:05:23] [Wrapper.py/INFO]: SurestTexas00 executed: /help VanillaClaims
Client.py - chatmsg: /transferclaim
Command.py - args: 
[]
Command.py - payload['args']: []
Command.py - ' '.join(payload['args']): 
[00:05:26] [Wrapper.py/INFO]: SurestTexas00 executed: /transferclaim 
Client.py - chatmsg: /adjustclaimblocks
Command.py - args: 
[]
Command.py - payload['args']: []
Command.py - ' '.join(payload['args']): 
[00:05:31] [Wrapper.py/INFO]: SurestTexas00 executed: /adjustclaimblocks 
[00:05:31] [Wrapper.py/ERROR]: Plugin 'com.suresttexas00.plugins.vanillaclaims' errored out when executing command: '<SurestTexas00> /{'callback': <bound method Main._adjustclaimblocks_comm of <vanillaclaims.Main instance at 0x7fb051bb4170>>, 'permission': 'vclaims.admin'}':
Traceback (most recent call last):
  File "/home/surest/Desktop/server test/wrapper/core/commands.py", line 61, in playerCommand
    command["callback"](payload["player"], payload["args"])
  File "wrapper-plugins/vanillaclaims.py", line 572, in _adjustclaimblocks_comm
    targetrname = str(SurestLib.makenamevalid(self, args[0], online=False, returnUUiD=False))
IndexError: list index out of range
Client.py - chatmsg: &6hello
[00:07:06] [Server thread/INFO]: <SurestTexas00> &6hello
Client.py - chatmsg: --seen SurestTexas00
[00:08:04] [Server thread/INFO]: <SurestTexas00> --seen SurestTexas00
[00:17:23] [Server thread/INFO]: [Server] World is saving... (Expect lag)
[00:17:23] [Server thread/INFO]: Saving...
[00:17:23] [Server thread/INFO]: commands.save.flushStart
[00:17:23] [Server thread/INFO]: ThreadedAnvilChunkStorage (worldx): All chunks are saved
[00:17:23] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[00:17:23] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[00:17:23] [Server thread/INFO]: commands.save.flushEnd
[00:17:23] [Server thread/INFO]: Saved the world
[00:17:23] [Server thread/INFO]: [Server] World saved.
[00:17:47] [Wrapper.py/ERROR]: Client Packet EOF (Packet was zero length, disconnecting)
Traceback (most recent call last):
  File "/home/surest/Desktop/server test/wrapper/proxy/client.py", line 665, in handle
    pkid, original = self.packet.grabPacket()
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 62, in grabPacket
    length = self.unpack_varInt()
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 95, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 379, in recv
    raise EOFError("Packet was zero length, disconnecting")
EOFError: Packet was zero length, disconnecting
[00:17:49] [Wrapper.py/INFO]: Starting new HTTPS connection (1): sessionserver.mojang.com
[00:17:49] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:17:49] [Wrapper.py/DEBUG]: "GET /session/minecraft/hasJoined?username=SurestTexas00&serverId=-11f9483385aa1666a738638101de83f978fc4543 HTTP/1.1" 200 1125
[00:17:49] [Wrapper.py/INFO]: Starting new HTTPS connection (1): api.mojang.com
[00:17:49] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:17:49] [Wrapper.py/DEBUG]: "GET /user/profiles/4f95325567754dc6a612fb4230588eff/names HTTP/1.1" 200 26
[00:17:49] [Wrapper.py/INFO]: SurestTexas00 logged in (UUID: 4f953255-6775-4dc6-a612-fb4230588eff | IP: 68.50.227.172)
[00:17:49] [Server thread/INFO]: SurestTexas00[/127.0.0.1:60668] logged in with entity id 361 at (3344.184574034524, 67.0, 2792.0658071568573)
[00:17:49] [Server thread/INFO]: SurestTexas00 joined the game
Client.py - chatmsg: hwelp
[00:17:54] [Server thread/INFO]: <SurestTexas00> hwelp
Client.py - chatmsg: /van
Command.py - args: 
[]
Command.py - payload['args']: []
Command.py - ' '.join(payload['args']): 
[00:22:19] [Wrapper.py/INFO]: SurestTexas00 executed: /van 
[00:23:07] [Wrapper.py/ERROR]: Client Packet EOF (Packet was zero length, disconnecting)
Traceback (most recent call last):
  File "/home/surest/Desktop/server test/wrapper/proxy/client.py", line 665, in handle
    pkid, original = self.packet.grabPacket()
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 62, in grabPacket
    length = self.unpack_varInt()
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 95, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "/home/surest/Desktop/server test/wrapper/proxy/packet.py", line 379, in recv
    raise EOFError("Packet was zero length, disconnecting")
EOFError: Packet was zero length, disconnecting
[00:23:09] [Wrapper.py/INFO]: Starting new HTTPS connection (1): sessionserver.mojang.com
[00:23:09] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:23:09] [Wrapper.py/DEBUG]: "GET /session/minecraft/hasJoined?username=SapperLeader2&serverId=-41cb739d15e7e111ffefe87544ce14058a64df5a HTTP/1.1" 200 1125
[00:23:09] [Wrapper.py/INFO]: Starting new HTTPS connection (1): api.mojang.com
[00:23:09] [Wrapper.py/DEBUG]: Setting read timeout to None
[00:23:09] [Wrapper.py/DEBUG]: "GET /user/profiles/5344845afb5e4a7d871bbe682e27e29c/names HTTP/1.1" 200 226
[00:23:09] [Wrapper.py/INFO]: SapperLeader2 logged in (UUID: 5344845a-fb5e-4a7d-871b-be682e27e29c | IP: 68.50.227.172)
[00:23:10] [Server thread/INFO]: SapperLeader2[/127.0.0.1:60740] logged in with entity id 456 at (3344.6041111857758, 67.0, 2791.2334135907586)
[00:23:10] [Server thread/INFO]: SapperLeader2 joined the game

```

**Comment by JasonBristol** [comment]
I don't see where `self.skins = {}` in ***proxy.py*** is ever set, I wonder if this is the root cause

**Comment by suresttexas00** [comment]
They were (used to be) set somewhere in the polling.

**Comment by JasonBristol** [comment]
Not entirely convinced it is uuid related, if it was the request would fail and we would see an error message, this looks more like a successful request was made, but the target was given your skin  not their own, maybe a swapped variable?

**Comment by suresttexas00** [comment]
Yea, likely.

**Comment by suresttexas00** [comment]
we should starting eye balling the playerlist packet...  and the uuid that is being sent with it... if a recall, player skin blobs are also sent in this packet.. and since my /van command is still not working (although the arguments are passed correctly now), the playerlist is defininately a suspect now... playerlist may even be affecting the spectator players.

**Comment by JasonBristol** [comment]
Interesting, I think you might be right, it would certainly explain things,
let me take a look at what changes wee made

**Comment by suresttexas00** [comment]
fixed by same fix as #47
